packages/block/src/header.ts

```diff
@@ -254,28 +254,36 @@ export class BlockHeader {
   }
 
   /**
-   * Returns the canonical difficulty for this block.
-   *
-   * @param parentBlockHeader - the header from the parent `Block` of this header
+   * @param parentBlock - The `parentBlock` where we wish to build upon
+   * @param timestamp - The `BN` timestamp which we should calculate the difficulty for
+   * @param number - The `BN` block number. Defaults to the parent blocks' `number` + 1
+   * @param common - The `Common` to use: defaults to the parent blocks` `Common`
```

Comment:
If we want to do this change to `static` we should really drop the `parent` here on all fronts and just call this `block`. There is no information lost here and this simplifies things and this parent doesn't make any sense any more in a static context.

Reply:
I have just pushed fd196353 but I am actually not sure anymore why I made this static in the first place. Since we always need a `header` (or a `block`) for this function, it makes more sense to remove `static`, just use the current header and let the caller specify the timestamp + (optionally) number/common?

Reply:
Yes, that makes sense, was just wondering too if this double structure with one static and one non-static method really makes sense